### PR TITLE
Clean up canvas event handling

### PR DESCRIPTION
**Pull request by eyuelt**
Clean up `event.preventDefault` and `event.stopPropagation` stuff

### CODE REVIEW

File: src/client/app/graph/canvas/canvas.component.ts [hunk 1]
```diff
@@ -17,19 +18,25 @@ export class CanvasComponent {
   // The last point seen during the pan that is currently in progress.
   lastPanPnt: Point = null; //TODO(eyuelt): make this nullable after TS2 update
   // Returns the bounding box of the canvas.
-  getBounds: {():ClientRect} = null;
+  getBounds: (() => ClientRect) = null;
+  // Returns whether the target of the given event is the canvas.
+  eventTargetIsCanvas: ((event:Event) => boolean) = null;
```
Comment (girum11) [inline src/client/app/graph/canvas/canvas.component.ts]: Make me private?

File: src/client/app/graph/canvas/canvas.component.ts [hunk 2]
```diff
@@ -17,19 +18,25 @@ export class CanvasComponent {
   // The last point seen during the pan that is currently in progress.
   lastPanPnt: Point = null; //TODO(eyuelt): make this nullable after TS2 update
   // Returns the bounding box of the canvas.
-  getBounds: {():ClientRect} = null;
+  getBounds: (() => ClientRect) = null;
```
Comment (girum11) [inline src/client/app/graph/canvas/canvas.component.ts]: Make me private?

File: src/client/app/service/utils/utils.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import {Injectable} from '@angular/core';
+
+@Injectable()
+export class Utils {
```
Comment (girum11) [inline src/client/app/service/utils/utils.ts]: I _really_ dislike naming things "Utils" -- it reminds me of variables named "data".

Any way to give this class a more meaningful name? Or maybe organize the functions by their usage? Or maybe both?

E.g. call this class "Events"?

File: src/client/app/service/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import {Injectable} from '@angular/core';
+
+@Injectable()
```
Comment (girum11) [inline src/client/app/service/utils/utils.ts]: I don't think you're injecting this class via DI at all; you should be able to remove this decorator.

Comment (girum11) [inline src/client/app/service/utils/utils.ts]: Also, since this isn't a service, maybe we should introduce some new folder for static functions?

**Comment by girum11** [comment]
LGTM

File: src/client/app/graph/canvas/canvas.component.ts (hunk 2 above)
Comment (eyuelt) [inline src/client/app/graph/canvas/canvas.component.ts]: Done

File: src/client/app/graph/canvas/canvas.component.ts (hunk 1 above)
Comment (eyuelt) [inline src/client/app/graph/canvas/canvas.component.ts]: Done

**Pull request merged**